azurerm/resource_arm_analysis_services_server_test.go

```diff
@@ -0,0 +1,379 @@
+	firewallRules2[0]["name"] = "test1"
+	firewallRules2[0]["range_start"] = "92.123.234.11"
+	firewallRules2[0]["range_end"] = "92.123.234.12"
+
+	config2 := testAccAzureRMAnalysisServicesServer_firewallSettings(ri, testLocation(), false, firewallRules2)
+
+	firewallRules3 := make([]map[string]string, 2)
+
+	firewallRules3[0] = make(map[string]string)
+	firewallRules3[0]["name"] = "test1"
+	firewallRules3[0]["range_start"] = "92.123.234.11"
+	firewallRules3[0]["range_end"] = "92.123.234.13"
+
+	firewallRules3[1] = make(map[string]string)
+	firewallRules3[1]["name"] = "test2"
+	firewallRules3[1]["range_start"] = "226.202.187.57"
+	firewallRules3[1]["range_end"] = "226.208.192.47"
+
+	config3 := testAccAzureRMAnalysisServicesServer_firewallSettings(ri, testLocation(), true, firewallRules3)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAnalysisServicesServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config1,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enable_power_bi_service", "true"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.#", "0"),
+				),
+			},
+			{
+				Config: config2,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enable_power_bi_service", "false"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.name", "test1"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_start", "92.123.234.11"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_end", "92.123.234.12"),
+				),
+			},
+			{
+				Config: config3,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "enable_power_bi_service", "true"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.name", "test1"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_start", "92.123.234.11"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.0.range_end", "92.123.234.13"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.name", "test2"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.range_start", "226.202.187.57"),
+					resource.TestCheckResourceAttr(resourceName, "ipv4_firewall_rule.1.range_end", "226.208.192.47"),
+				),
+			},
+		},
+	})
+}
+
+// Currently unsure how to test it as the email addresses need to exist in the AD
```

Comment:
Would it be possible to create the user's via the azuread provider & use them here?

Reply:
Talked about this with @tombuildsstuff and he proposed to work with a env var which points to a email address in the azure ad. That okay for you? Then everyone can test in their own azure tenant.

Reply:
You now need to specify two email addresses as env variables to run this test. Unfortunately it's not working as expected -> test fails but i don't know why.